src/ripple/app/tx/impl/CancelCheck.cpp

```diff
@@ -104,22 +106,18 @@ CancelCheck::doApply ()
     if (srcId != dstId)
     {
         std::uint64_t const page {(*sleCheck)[sfDestinationNode]};
-        TER const ter {dirDelete (view(), true, page,
-            keylet::ownerDir (dstId), checkId, false, false, viewJ)};
-        if (! isTesSuccess (ter))
+        if (! view().dirRemove (keylet::ownerDir(dstId), page, checkId, true))
         {
             JLOG(j_.warn()) << "Unable to delete check from destination.";
-            return ter;
+            return tefBAD_LEDGER;
         }
     }
     {
         std::uint64_t const page {(*sleCheck)[sfOwnerNode]};
-        TER const ter {dirDelete (view(), true, page,
-            keylet::ownerDir (srcId), checkId, false, false, viewJ)};
-        if (! isTesSuccess (ter))
+        if (! view().dirRemove (keylet::ownerDir(srcId), page, checkId, false))
```

Comment:
This `keepRoot` also got swapped

Reply:
Good catch, again. I must have been working late when doing this -- apologize for the simple errors. It's pretty evident there are some missing unit tests for these cases though with respect to preserving the original directory as you've pointed out.

Reply:
Addressed in https://github.com/JoeLoser/rippled/commit/f810d6742dc4139aeec09556bb5bd054ec0ea0da